### PR TITLE
Adding tutorials

### PR DESCRIPTION
**Pull request by jessegoodier**
### Proposed changes
Adding tutorials for database/UDP and App Protect using an OpenAPI policy

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: docs/content/tutorials/database-ingress/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+In the [values-plus.yaml](values-plus.yaml) there is a section for globalConfiguration which will tell NGINX to listen on non-standard ports- see the redis section in this and the extra services.
+
+You will need to add the NGINX Helm repository to use it:
+
+```
+helm repo add nginx-stable https://helm.nginx.com/stable
+helm repo update
+helm install plus nginx-stable/nginx-ingress --namespace nginx-ingress -f values-plus.yaml
+```
+Note that the values file expects that you have already created a secret named regcred in the nginx-ingress namespace.
```
Comment (gallarda) [inline docs/content/tutorials/database-ingress/README.md:22]: You may want to go ahead and show how to create the "regcred" secret here.

**Review by jasonwilliams14 — COMMENTED** [review]
Made some general comments and suggestions.
I think we should restructure the folder layout, make it flat
database-ingress (for top level)
-----redis
-----mysql

etc.

but for all sections.

File: docs/content/tutorials/database-ingress/README.md (hunk 1 above)
Comment (jessegoodier) [reply docs/content/tutorials/database-ingress/README.md:22]: agree, thanks for reviewing!

File: docs/content/tutorials/database-ingress/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Database Load Balancing
+
+NGINX Ingress Controller supports load balancing arbitrary TCP and UDP services using a Custom Resource Definition called a TransportServer.
+
+## Usage
+
+This directory contains a sample using redis; this can easily be adapted to other services.
+
+## Installing NGINX Ingress Controller
```
Comment (brianehlert) [inline docs/content/tutorials/database-ingress/README.md:9]: We should set the pattern of referring to one set of installation documents.
That frees the tutorial to focus on describing the use case and the configuration of NIC and any backing service.

If coffee-tea is used, then we should refer to one set of instructions to deploy that service as well.

If we need the enhance the installation doc then we should.
If we need to call out specific gotchas to make sure the customer has done something, that is fine to be here.

File: docs/content/tutorials/database-ingress/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+Note: Alternatively, you can build the ingress controller following the [building the ingress controller documentation](https://docs.nginx.com/nginx-ingress-controller/installation/building-ingress-controller-image/).
+
+In the [helm-files/values-plus-redis.yaml](helm-files/values-plus-redis.yaml) there is a section for globalConfiguration which will tell NGINX to listen on non-standard ports- see the redis section in this and the extra services. This is also possible with the manifest yaml files.
+
+Create the secret to pull from the private registry:
+
+```
+kubectl create secret docker-registry regcred --docker-server=private-registry.nginx.com --docker-username=<NGINX Plus JWT Token> --docker-password=none -n nginx-ingress
+```
+Complete documentation for creating the secret is here: [https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-helm/]
```
[the remark this answers is not in the document]
Comment (brianehlert) [reply docs/content/tutorials/database-ingress/README.md:22]: We should refer to our primary docs and not rehash common processes across tutorials.

If the links need to be a pop-out for ease, we should do that.

[the remark this answers is not in the document]
Comment (jasonwilliams14) [reply docs/content/tutorials/database-ingress/README.md:22]: @ciarams87 Does it make sense to add/highlight this section to say "If you are using JWT Secret....go here. If you are an existing customer and pulling from your private repo, go here?" That way, both avenues are covered?

yes, pretty links are nice indeed and add nice touch to documentation that customers appreciate.

**Comment by lucacome** [comment]
@jessegoodier @ashutoshpradhan-f5 @nginxinc/nginx-docs what is the status of this PR?

**Comment by brianehlert** [comment]
Jesse has left the company.  He is welcome to continue to contribute.
The tutorial should be contributed to the tutorials branch

**Comment by jputrino** [comment]
Should we close the PR until such time as Jesse (or anyone else) decides to revisit the topic?

**Comment by brianehlert** [comment]
Probably wise at this point.
We have other tutorials from Jason.  Those should be going into the branch.

**Pull request closed without merging**